src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/AddPartition.sql

```diff
@@ -30,6 +30,13 @@ BEGIN
     DECLARE @createdDate DATETIME2(7) = SYSUTCDATETIME()
     DECLARE @partitionKey INT
 
+    SELECT @partitionKey = PartitionKey
```

Comment:
This will reduce the occurance but won't elemenate it. You will have to hold a xLock to block any insertions after the check and before the insert.

Or you can do what STOW does in the c# store layer. Handle the conflict error and throw the exception

dicom-server\src\Microsoft.Health.Dicom.SqlServer\Features\Store\SqlIndexDataStoreV10.cs
catch (SqlException ex) when (ex.Number == SqlErrorCodes.Conflict

Reply:
![image](https://user-images.githubusercontent.com/3883800/207127444-cdcf9c66-7de6-48e9-be8f-662238b3fd34.png)

Reply:
Looks like I missed it. Makes sense